src/pact.ts

```diff
@@ -146,10 +150,19 @@ export class Pact {
     this.finalized = true;
 
     return this.mockService.writePact()
-      .then(() => this.server.delete())
-      .catch((err: Error) => {
-        return Promise.all([this.server.delete(), Promise.reject(err)]);
-      });
+      .then(() => logger.info("Pact File Written"), (e) => {
+        console.log("Pact File Write Failure");
+        return Promise.reject(e);
+      })
+      .then(() => new Promise<void>((resolve, reject) => this.server.delete().then(() => resolve(), (e) => reject(e))))
```

Comment:
This is a bit hard to read/reason - worth extracting?

Reply:
yeah, wanted to add a TODO to it.  The issue here is that it seems that q promises are not interchangeable with regular Promises from a typing perspective.  I need to fix that so that we don't have this horrible issue of typing issues, but that's outside of the realm of this PR.  I really loathe the fact that JS promises don't have the `finally` just yet or any of the other niceties that q provides.

Anywho, this is the only way I could do it without having the TS compiler complaining about types not matching and having the promise actually wait for the server to delete since before it wasn't waiting for it. this will be temporary.